Mathlib/NumberTheory/LSeries/DirichletContinuation.lean

```diff
@@ -307,3 +309,91 @@ theorem completedLFunction_one_sub {χ : DirichletCharacter ℂ N} (hχ : IsPrim
 end IsPrimitive
 
 end DirichletCharacter
+
+/-!
+### The logarithmic derivative of the L-function of a Dirichlet character
+
+We show that `s ↦ -(L' χ s) / L χ s + 1 / (s - 1)` is continuous outside the zeros of `L χ`
+when `χ` is a trivial Dirichlet character and that `-L' χ / L χ` is continuous outside
+the zeros of `L χ` when `χ` is nontrivial.
+-/
+
+namespace DirichletCharacter
+
+open Complex
+
+section trivial
+
+variable (n : ℕ) [NeZero n]
+
+/-- The function obtained by "multiplying away" the pole of `L χ` for a trivial Dirichlet
+character `χ`. Its (negative) logarithmic derivative is used to prove Dirichlet's Theorem
+on primes in arithmetic progression. -/
+noncomputable abbrev LFunctionTrivChar₁ : ℂ → ℂ :=
+  Function.update (fun s ↦ (s - 1) * LFunctionTrivChar n s) 1
+    (∏ p ∈ n.primeFactors, (1 - (p : ℂ)⁻¹))
+
+lemma LFunctionTrivChar₁_apply_of_ne_one {s : ℂ} (hs : s ≠ 1) :
+    LFunctionTrivChar₁ n s = (s - 1) * LFunctionTrivChar n s := by
+  simp only [ne_eq, hs, not_false_eq_true, Function.update_noteq]
+
+lemma LFunctionTrivChar₁_apply_one_ne_zero : LFunctionTrivChar₁ n 1 ≠ 0 := by
+  simp only [Function.update_same]
+  refine Finset.prod_ne_zero_iff.mpr fun p hp ↦ ?_
+  rw [sub_ne_zero, ne_eq, one_eq_inv]
+  exact_mod_cast (Nat.prime_of_mem_primeFactors hp).ne_one
+
+/-- `s ↦ (s - 1) * L χ s` is an entire function when `χ` is a trivial Dirichlet character. -/
+lemma LFunctionTrivChar₁_differentiable : Differentiable ℂ (LFunctionTrivChar₁ n) := by
```

Comment:
Maybe `differentiable_LFunctionTrivChar1`, for consistency with `deriv_LFunction[xxx]` below?